module3/homework/vectorOfSharedPointers/vectorFunctions.cpp

```diff
@@ -0,0 +1,52 @@
+#include "vectorFunctions.hpp"
+
+#include <iostream>
+
+std::vector<std::shared_ptr<int>> generate(int count)
+{
+    std::vector<std::shared_ptr<int>> vec;
+    vec.reserve(count);
```

Comment:
You can also reserve memory in constructor:
```suggestion
    std::vector<std::shared_ptr<int>> vec(count);
```